Add lightning-privacy-splice-panel review to bitcoin-review-podcast (#635)
bitcointranscripts/bitcointranscripts: bitcoin-review-podcast/lightning-privacy-splice-panel.md

Reviewer: Change ~~Cashew~~ to `Cashu` in this sentence. There are two occurrences.

Reviewer: ```suggestion
Like it's a, it caught me by surprise at how fast, especially because of Cashu, the public development of Cashu happened, like how sort of like ahead this is and how usable it already is.
```

Reviewer: Something is wrong with Github. It doesn't show its original sentence and diff.

Reviewer: ```suggestion
But Nevertheless, it is very exciting to see if it does come to fruition because it's been in the works for a very long time as in other names as SIGHASH_NOINPUT, many other things like that.
```
It should be `SIGHASH_NOINPUT`

Reviewer: ```suggestion
Oh, Rijndael too.
```

Reviewer: https://substack.bitcoin.review/p/bitcoin-review-podcast-br095-op_next
`Rijndael`, not ~~Rindell~~

Reviewer: ```suggestion
Rijndael explains everything so well to me.
```

Reviewer: ```suggestion
That JoinMarket doesn't have like a very dumbed down UI.
```

Reviewer: ```suggestion
Fortunately, people like the non-Full-RBF people probably just learned now that why you need Full-RBF because the fees are high.
```

Reviewer: ~~non-FooRBF~~ people probably just learned now that why you need ~~FooRBF~~

---
title: 'Lightning Privacy & Splice Panel'
transcript_by: 'kenda-sketch via review.btctranscripts.com'
media: 'https://www.youtube.com/watch?v=SjLPbjs9LkI'
date: '2023-05-12'
tags:
  - 'splicing'
  - 'rv-routing'
  - 'offers'
  - 'privacy-enhancements'
speakers:
  - 'NVK'
  - 'Bastien Teinturier'
  - 'Jeff Czyz'
  - 'Dusty Dettmer'
  - 'Tony Giorgio'
  - 'Vivek'
---
## Housekeeping

> **NVK:** 00:01:17

Reviewer: ```suggestion
**NVK:** 00:00:37

Hello and welcome to the bitcoin.review podcast where we explore developments and projects with the people who actually make them happen. The show is supported by Pod 2.0 sat streaming and CoinKite. If you're a new listener, I'm NVK. I run CoinKite where we've been helping people secure their Bitcoins for over a decade. We make the Coldcard and fun products like the BlockClock. You can find more information about it on coinkite.com. Hello and welcome back to the bitcoin.review podcast. My voice is back and we did manage to bore everyone to death on the last episode. The list was long and prosperous. 
```

Today we're going to be talking about `Lightning` privacy, splicing and other very cool things with an absolute amazing panel here.

> <!-- Just before that, I have a couple of things that I want to set out some updates.

Reviewer: ```suggestion
Just before that, I have a couple of things that I want to set out some updates.
```

> So OpenSats is now taking grants for a Bitcoin application, sorry, taking grant applications for Bitcoin projects and Nostra projects and `Lightning` projects.

Reviewer: ```suggestion
So OpenSats is now taking grants for a Bitcoin application, sorry, taking grant applications for Bitcoin projects and Nostr projects and `Lightning` projects.
```

> And there is about five million for an OSTER, five million for Bitcoin.

Reviewer: ```suggestion
And there is about five million for Nostr, five million for Bitcoin.
```

So do apply.

> I joined as a board member and Gigi joined as the Ops person and we're gonna get those SATs flying.

Reviewer: ```suggestion
I joined as a board member and Gigi joined as the Ops person and we're gonna get those sats flying.
```

So let's get that funding going.
The other thing is for cold card users, we have merged Taproot and BSMS with that.

> So expect a separate edge release. -->

Reviewer: Why are you placing texts inside the `<!--` and `-->` brackets?  Would it hide the text from being displayed  when rendered?

> So expect a separate edge release. -->

Reviewer: ```suggestion
So expect a separate edge release. 
```

## Guest introductions

**NVK:** 00:02:14

So without further ado, guys, I have some awesome guests here.
Let's start with some intros.
Jeff, welcome back.

**Jeff Czyz:** 00:02:25

Hey, how's it going?
So yeah, my name is Jeff.
I am a developer at Spiral.
I've been working in the Bitcoin space for about four years now and my current project is the `Lightning` Development Kit implementation of `Lightning`, it's more like a SDK.

**NVK:** 00:02:41

Awesome.

> Tony, pickle Tony.

Reviewer: ```suggestion
Tony, tickle Tony. 
```
To me, it sounds more like "tickle" than "pickle", which matches the YouTube transcript. Since I can’t find either nickname online, the host was likely just being playful. It's not his nickname. WDYT?

https://www.youtube.com/watch?v=SjLPbjs9LkI&t=163s
<img height="100" alt="Image" src="https://github.com/user-attachments/assets/67022819-465c-4423-aee4-320dbae7c4b7" />

**Tony Giorgio:** 00:02:45

Hey guys, this is Tony.
`Lightning` dev, one of the co-founders of Mutiny with Ben and Paul Miller.
Yeah, I've been talking about lighting privacy for a while, so looking forward to this.

**NVK:** 00:02:55

Very cool.
Tbast.

**Bastien Teinturier:** 00:02:58

Hey, I'm Tbast.
I've been working at Async on the lighting specification and our implementation Eclair and also our Phoenix wallet.
And I've done a lot of things around lighting privacy as well, like blinded paths.
So excited to be talking about that here.

**NVK:** 00:03:14

Yeah, we did an episode, the `Lightning` panel, episode 15.
So for people interested on this, they should definitely go check out that episode.
Dusty Damon.

**Dusty Dettmer:** 00:03:24

Hey, what's up?

> <!-- Dusty Damon.

Reviewer: Why add `<!-- ` ?

Not that it matters.

**NVK:** 00:03:30

I know.

> I, you know, it's people haven't settled if it's like demon or Damon for, for the demons.

Reviewer 2: ```suggestion
I, you know, it's people haven't settled if it's like demon or Daemon for, for the demons.
```

**Dusty Dettmer:** 00:03:37

It's like proven work.

> If you can like, you know, pronounce Damon, then you can find me on Twitter and also spell it.

Reviewer 2: ```suggestion
If you can like, you know, pronounce Daemon, then you can find me on Twitter and also spell it.
```

So good luck with that.

**NVK:** 00:03:47

> And so Dusty, what do you do? -->

Reviewer: ```suggestion
And so Dusty, what do you do? 
```
Why add the bracket?

**Dusty Dettmer:** 00:03:49

I'm an independent `Lightning` Protocol Engineer.
I've been working on splicing for the last two years now.

**NVK:** 00:03:55

So does that mean we ship?

**Dusty Dettmer:** 00:03:57

Shipping, like right now.

> Me and Team Best, we got to get Interop going soon.

Reviewer: ```suggestion
Me and Tbast, we got to get Interop going soon.
```

Soon.

**NVK:** 00:04:04

Two weeks?

**Dusty Dettmer:** 00:04:05

Two weeks.

**NVK:** 00:04:06

OK.
And Vivek, Seared Salmon, welcome back.

**Vivek:** 00:04:12

Thank you.

> Just, you know, VP of BD at CoinKite and recovering Greenlight and CLN shoe.

Reviewer 2: ```suggestion
Just, you know, VP of BD at CoinKite and recovering Greenlight and CLN show.
```

> Just, you know, VP of BD at CoinKite and recovering Greenlight and CLN show.

Reviewer: ```suggestion
Just, you know, VP of BD at CoinKite and recovering Greenlight and CLN shill.
```

Reviewer: Vivek previously("recovering") worked at Blockstream and promoted("shill") those two projects.

**NVK:** 00:04:22

Nice to have you, sir.
So props to Vivek for putting this panel episode together. It's kind of cool.

## Current state of LN privacy

**NVK:** 00:04:29

So guys, where do we start here?
This is a hairy long, big topic.
So, how about, why don't we sort of like maybe cover a bit of a primer on, on the state of things?
Like Where are we in `Lightning` in terms of like things that are privacy related that are either just shipped or about to ship?
Let's not talk about things that are going to come in months from now still.
Like what's going on right now?
Who wants to take that?

> Go ahead, T-Best.

Reviewer 2: ```suggestion
Go ahead, Tbast⁠,.
```

**Bastien Teinturier:** 00:05:51

Receiver privacy is bad for many reasons.
First of all, if you when you receive a payment at the cryptographic level, because of onion routing, the guy just before you shouldn't be able to infer that you are the final recipient.
But if you're not careful about adding `CLTV` expiry deltas on top of the current block height, it's actually very easy to infer that the next guy is actually the recipient.
Also when you are using `unannounced channels`, oh actually this has become better.
A few months before when you were using `unannounced channels`, your invoices were leaking your `unannounced channels` all across everywhere because people just pasted invoices directly on Twitter and once you've doxed your channel, you've doxed it forever, or you have to pay on-chain fees to open a new one, this has become better since we introduced a CID aliases, which as you give an alias to a channel, which is much better, but it doesn't hide the public key of a recipient.
And this is getting much better with Blended Path.
With Blended Path, you're finally getting good privacy with some trade-offs on potential usability.

## Blinded paths

**NVK:** 00:06:59

> How do Blended Paths work?

Reviewer 2: ```suggestion
How do Blinded Paths work?
```

**Bastien Teinturier:** 00:07:01

> Okay, so Blended Paths are if you, I guess everyone has a vague idea of how rendezvous routing works in Onion networks, where the receiver, when you want to receive a message, instead of directly giving your address or your Node ID, you find paths to yourself that take a few hops to reach you.

Reviewer 2: ```suggestion
Okay, so Blinded Paths are if you, I guess everyone has a vague idea of how rendezvous routing works in Onion networks, where the receiver, when you want to receive a message, instead of directly giving your address or your Node ID, you find paths to yourself that take a few hops to reach you.
```

So people who are sending payments to you only learn the identity of a few introduction points that you chose on the network so you can choose them however you want to make sure that people cannot infer that then you are the recipient.
So algorithmically speaking and in order to be able to receive, to make the most of your incoming liquidity.
This isn't trivial to correctly choose your blinded paths.
And that's something that we are experimenting with a lot.
And there's a lot of cool stuff to do to make sure that this doesn't impact usability too much.
But in theory, once you're able to choose routes that still allow you to have enough income and liquidity, you're able to make people enter a route towards you far away from you or at least in different places in the network so that they just have no way of figuring out what node ID you are and it's even better if you are a wallet user that doesn't have any public announced channel because then no one should see your node ID anywhere.
You have no reason to advertise anything.

**Tony Giorgio:** 00:08:32

> With raw network routing, you hinted at it, but who you select as your introductory nodes is important too.

Reviewer 2: ```suggestion
With rendezvous routing, you hinted at it, but who you select as your introductory nodes is important too.
```

So, I wouldn't say like, you know, we're safe yet.
I mean, there are chain analysis companies looking at `Lightning`.

**NVK:** 00:15:29

But personally, just I'll say just assume it's being captured.
I mean, like that's the best way of looking at this.
But even then, I mean, like the privacy leaking graph is still pretty shallow, right?
It's not like on chain stuff.
Like you still have like a much better privacy set.
Let's put it this way on `Lightning`, right
if I am using my, you know, my Breeze or my Phoenix wallets, which is like, you know, just kind of like not fully custodial, but also not like your own node, It's kind of like in the middle there.
And I'm paying for a beer at the Bitcoin park.
And they're using their whatever, like BTC pay server there.
You know, unless somebody is really, really trying to get me, the chances of anybody seeing that that's what happened are pretty much lower than if I was using on-chain transactions, right?

**Vivek:** 00:16:21

I don't know if it's much lower, but I'd say it's lower and there's definitely aggregators like Tony was saying, and that's always been the concern.

> I remember when Ellen Big showed up, everyone was like, oh, these guys are the feds or there's some other agency.

Reviewer: ```suggestion
I remember when LNBIG showed up, everyone was like, oh, these guys are the feds or there's some other agency.
```

But you know, you just kinda, you're happy to get that inbound liquidity at the end of the day.
I actually want to go higher level and just get our principles first, fundamentals established for the `Lightning` Network, the challenges for the privacy.

> I think I've heard T-Best and Tony explain it for those who haven't checked out their contributions, You should definitely check it out.

Reviewer 2: ```suggestion
I think I've heard ⁠Tbast and Tony explain it for those who haven't checked out their contributions, You should definitely check it out.
```

> T-Best has an amazing chapter in the Mastering `Lightning` book, which I recommend nonstop.

Reviewer 2: ```suggestion
⁠Tbast has an amazing chapter in the Mastering `Lightning` book, which I recommend nonstop.
```

And then Tony just sprung onto the scene, I think in 2021 with a massive privacy article describing issues with payment hashes, hash correlation, long paths.

> So from a high level approach, Tbast, can you explain how, is it Bolt1 or something, the init messages with noisexk, basically how the sender is always unknown and the receiver used to be known and the challenges with that in regards to `unannounced channels` which were mistakenly called private channels back in the day.

Reviewer: ```suggestion
So from a high level approach, Tbast, can you explain how, is it Bolt1 or something, the init messages with Noise_XK, basically how the sender is always unknown and the receiver used to be known and the challenges with that in regards to `unannounced channels` which were mistakenly called private channels back in the day.
```

Our goal with `Lightning`, once you are inside `Lightning` and sending `Lightning` payments, our goal is that any node that is in the route can only infer who a previous node was and who the next node was, and nothing more than that.
Because that's something we cannot avoid.
That's how they receive a message from someone so they know that it came from that direction and they're sending a message to someone so they know it's going that other direction.
But we want to remove anything else they could use to track you.
So right now, we're using the same payment hash across the whole route, which is an issue.
PTLCs will fix that.
That also means that the sender, for example, should not be able to infer who the recipient is based on only potentially an invoice.
The issue right now is that in the Bolt11 invoice, that when it doesn't yet support SCID aliases, is that you are leaking both your node ID, which you don't actually really have to, but it's a bit inconvenient to hide it with Bolt11 currently, and exactly the channels that tell people how to reach you and who you are connected to.

> So this part is more about the intersection between `Lightning` and Unchain.

Reviewer 2: ```suggestion
So this part is more about the intersection between `Lightning` and Onchain.
```

They are completely orthogonal and operating at different layers of the stack and require very different solutions.
And to be honest, I think that we have good ideas on how to somewhat easily fix at the protocol level.
Almost everything that happens once you are in `Lightning` and sending payments, it's much harder to find changes that are only at the protocol level and that will fix the potential privacy leaks when you are doing on-chain to off-chain stuff.
Does that answer what you wanted or is there still something that I missed?

## `Lightning` privacy challenges

**Vivek:** 00:19:57

No, that's pretty good.
I just mainly wanted to touch on exactly what you said, how we're kind of piggybacking off of all the on-chain activity, with the SCIDs. And maybe I can ask Tony about, the alias change that happened for ZeroConf and Maybe some goodness there.

**Tony Giorgio:** 00:20:16

Yeah, I mean, that was huge.
I know my article, abytesjourney.com slash `Lightning` privacy, I can put it in the show notes.
I wrote that in like 2011.

**Vivek:** 00:20:29

> 2021.

Reviewer 2: ```suggestion
 2021
```

## Routing and LSPs

**Jeff Czyz:** 00:22:49

T-BAST, would you say then that like for that scenario, the LSP wouldn't be the introduction point on the blinded route?
And if so, how would a mobile user choose a different introductory point?

**Bastien Teinturier:** 00:23:02

> Either the mobile user just syncs a local neighborhood, for example, every node that is two hubs away from them, if they're only connected to one LSP, they cannot just sync it from that LSP if they don't trust that LSP because the LSP will just filter out everything else.

Reviewer 2: ```suggestion
Either the mobile user just syncs a local neighborhood, for example, every node that is two hops away from them, if they're only connected to one LSP, they cannot just sync it from that LSP if they don't trust that LSP because the LSP will just filter out everything else.
```

So the act of gossiping currently in Bolt7 is not very privacy preserving.
Now I think there are some proposals out there, one for zero knowledge proofs to be used for this.
So you kind of say, I have some on-chain balance.
I'm not going to say how much exactly it is, but I can at least prove to you that I do.

**NVK:** 00:26:51

It's like a range proof kind of thing?

**Jeff Czyz:** 00:26:53

I believe so.
I don't know the exact details honestly, but that's essentially the idea.

**Vivek:** 00:26:59

This is a Rusty's idea for the ownership proof or something like that?

**Jeff Czyz:** 00:27:04

I'm not sure if it's the same, but there are a few proposals out there.
There's like Gossip 1.5, which I believe that might be you're referring to.
But I could be mistaken there.
Someone else could correct me and then there's more of a long-term gossip B2 where we want to maybe be a little more privacy preserving than that proposal would allow.
I don't know if anyone else wants to jump in on that, probably a little more details.

## Speed of development

**NVK:** 00:27:27

> T-BAST, you wanted to bring up something about the features that are sort of slow to?

Reviewer 2: ```suggestion
Tbast , you wanted to bring up something about the features that are sort of slow to?
```

So I'm literally in the process of rewriting things for those new changes, just like Jim was talking about.

**NVK:** 00:33:43

That's when the bike shedding begins.

**Dusty Dettmer:** 00:33:45

Yes, exactly.

**Jeff Czyz:** 00:33:48

I remember listening to your talk at Bitcoin++ last year on splicing and now a year later, it's getting a lot of traction now, but there's a lot of movement too.
And what is the final specification?

## Why an additional node implementation? (`LDK`)

**NVK:** 00:34:00

So Jeff, here's a related question, a bit, I would say a bit tangent, but so if we already had say three node implementations, why create a fourth and add one more to the sort of like the dynamic of getting features out.
Right, because you know, every time you add one more player that has enough market share, it adds more complication.
I love `LDK`.
I'm not like, that's not a criticism of `LDK`.
I'm just curious, what was the rationale for that?
Aside from just like, we wanna do things in a certain different way.
So it's worth adding one more implementation to the market.

**Jeff Czyz:** 00:34:39

Well, the more the merrier, right?

**Vivek:** 00:34:42

> Jeff, don't let the C programmer bully the rest community like that.

Reviewer: ```suggestion
Jeff, don't let the C programmer bully the Rust community like that.
```

He joked around using "crust" later in #L669

**Jeff Czyz:** 00:34:45

Yeah.

**NVK:** 00:34:47

No, it's crust now because rust is a trademark.
So I'm calling it crust.

**Jeff Czyz:** 00:34:53

I think that's been resolved, but yeah.
So, we had a different approach, right?
So we saw a sort of gap in mobile wallet usages of `Lightning`.
Or like there were many usages that were very custodial, you know, they might be relying on servers and not having an easy path for Bitcoin wallets to add `Lightning` was something we saw as a gap in the ecosystem.
So that was the primary reason I believe that `LDK` was sort of pitched in that direction, I suppose.

> We do use Rust compared to other implementations like C and Go, and I guess Eclare is Scott's style, is it?

Reviewer 2: ```suggestion
We do use Rust compared to other implementations like C and Go, and I guess Eclair is Scott's style, is it?
```

> We do use Rust compared to other implementations like C and Go, and I guess Eclair is Scott's style, is it?

Reviewer: ```suggestion
We do use Rust compared to other implementations like C and Go, and I guess Eclair is Scala, is it?
```

So they're not sort of like so much on top of each other because there's only so many customers too for now.
Sorry, T-BAST.

**Bastien Teinturier:** 00:36:38

From the spec point of view, I'm really happy that there's a fourth implementation because the way, the best, the ideal thing when you start working on a new big spec feature is to have another implementation, start working on it almost at the same time as you, because doing it on your own means that whenever someone will come one year later, they will want to change everything potentially for good reason, but you're wasting a bit of time.
But when we only had three implementations and since people have different short term focus and everything we're doing takes a lot of time, so people cannot be working on all the things at the same time, we have a better chance at finding two implementations that work on the same stuff for almost each big feature that we are working on.
And I really think this helps, for example, a very simple example is Taproot, because right now Taproot is not a priority for Eclair and Core `Lightning`.
It is a huge priority for LND.

> So I think they're very thankful that `LDK` has found time to actually really work on it, give them feedback and make sure that they have two implementations working together on that specification because otherwise L&D would have been blocked because they were the only one working on that and it's a big chunk of work and Core `Lightning` and Eclair thought it was more important to start with things like splicing and dual funding before moving on to Taproot.

Reviewer 2: ```suggestion
So I think they're very thankful that `LDK` has found time to actually really work on it, give them feedback and make sure that they have two implementations working together on that specification because otherwise LND would have been blocked because they were the only one working on that and it's a big chunk of work and Core `Lightning` and Eclair thought it was more important to start with things like splicing and dual funding before moving on to Taproot.
```

Yeah.
Putting descriptions in memos, I think, was a bad decision for `Lightning`.
And not just putting it there, requiring it.
Requiring that field or a description hash is required and the reason that and at the time it was a good idea, right?
Because you want to know that when you pay an invoice that it's gonna be for a specific order like you could have an order number It could have a list of the items in the description.
It's practical for it to be there, but what is ending up happening, especially when we have large custodians on the `Lightning` Network as more chain analysis starts aggregating more data, but the invoice is being collected, you start to now see you start to attach the receivers pub key to the reason for payment.
And that gets if that's getting shared and aggregated around, you can now start correlating people with the reasons that they were paid.
And that's not good.
In my opinion.

**NVK:** 00:46:21

Also, it's pretty annoying, to be honest, as a user, like, stop asking me, forcing me to put a note.
Like, I don't want to put a note.
Like, it's the same with wallets that require an amount.
Like, I don't want to, like, just leave me alone.

> You know, it's weird.

Reviewer: ```suggestion
It's weird.
```

> Cause like, you know, I'm a base layer boomer.

Reviewer: ```suggestion
Cause I'm a base layer boomer.
```

So yeah, I don't know,

**Bastien Teinturier:** 00:49:07

to be honest.

**Tony Giorgio:** 00:49:08

Yeah, I'm very concerned about `Lightning` usage on Nuster for both the Zaps aspect and also the custodial integrations like Wallet Connect.
You know, there's a side of it where it's like, we can be up here preaching about lighting privacy all day long and then someone connects their node up with AMBOSS and shares the balance.
Like, we can't stop that on a protocol level.
And all we can do is educate and tell people why it's dumb, why you why you shouldn't do that, why you shouldn't use custodians.
We can preach this all day long, but then someone goes and

**NVK:** 00:49:44

do it.
I mean, we can't say anything like saying won't change behavior, offering tools that fix the behavior that are easier to use will fix the problem.
Right.
Like and this is why I'm bringing this up because you know like when you actually look at like the transaction graph I mean like it looks like Nostr's like what like 70% of all the `Lightning` transactions at least was the peak.

**Vivek:** 00:50:08

Yeah, I think Ben Carman did some research.

> Yeah, it's 70% of all Nostr's apps were Wallet of Satoshi or something like that, right, Tony?

Reviewer: ```suggestion
Yeah, it's 70% of all Nostr zaps were Wallet of Satoshi or something like that, right, Tony?
```

We gotta remove the barriers, make it easy to do it correctly.
And then we can evangelize doing it correctly to people and like actually have that movement get some team.

**Jeff Czyz:** 00:53:13

Yes, it's important just to provide that option.
People will use custodial services, but they need to have the opportunity to self-custody.

**NVK:** 00:53:20

Well, I mean, being very realistic, I mean, like `Lightning` doesn't scale past a certain amount too, right?
So base layer takes us from like, you know, having cuck money, right?
Having a central bank to not having a central bank, right?
Like, and, and, you know, like it's more akin to Fedwire.
Hey, great.
We can now send each other a million dollars for like reasonable fees and nobody can stop us.

> Right Now we can't buy coffee on Bayes layer as people are, I don't know, somehow people are surprised that the fees are going to be high, you know, like sure it's the DGN right now who are clogging up the pipes, but like soon is going to be just the population of the world.

Reviewer: ```suggestion
Right Now we can't buy coffee on Bayes layer as people are, I don't know, somehow people are surprised that the fees are going to be high, you know, like sure it's the degens right now who are clogging up the pipes, but like soon is going to be just the population of the world.
```

Reviewer: Youtube auto-generated subtitles
<img width="2017" height="377" alt="Image" src="https://github.com/user-attachments/assets/4f3d6f03-965b-421c-b012-8fc8b39abd86" />

I think like, it's, It's just like you said, there's like three different things there.
It's a whole complex rabbit hole.
But simply put, it's the ability to resize `Lightning` channels without closing them.
But what's become apparent over time is that unlocks a whole lot of things.
What you're doing is you're essentially creating fungibility from on-chain Bitcoin to `Lightning`, which changes the entire game.
At first, it's like, ah, just resizing channels, whatever.
But then literally everything changes about it, including there are some theoretical concepts now about changing how coin joins work to integrate splicing and perhaps make those better, get some like stenographic techniques going.
So like literally everything's changing.
There's sort of like a laundry list and depending on like which part you wanna talk about first we can go into the individual one.

**NVK:** 00:55:43

So first question is how long do he ships?

**Dusty Dettmer:** 00:55:48

Two weeks.

**NVK:** 00:55:50

Before we get into this, two weeks.

**Vivek:** 00:55:54

Dusty shipped it for himself, right?
I don't know if he's revealed like any of the script, but he did point to a mainnet transaction.

> And I believe T-Best also has done that.

Reviewer 2: ```suggestion
And I believe ⁠Tbast also has done that.
```

of this.

**Dusty Dettmer:** 00:56:42

Yes, let's plant my flag.

> I did the first splice on Chain last year in May and I finished my Splicing Limitation around October.

Reviewer: ```suggestion
I did the first splice on Chain last year in May and I finished my specification around October.
```
https://www.youtube.com/watch?v=SjLPbjs9LkI&t=3406s

<img width="2042" height="342" alt="Image" src="https://github.com/user-attachments/assets/e5f6808f-b96c-4c39-b442-e73115123126" />

Sounds more like "specification" WDYT? @0tuedon

> But like things in `Lightning` aren't normally considered done until two Limitations do it, sort of like the way that things do it, call it interop.

Reviewer: ```suggestion
But like things in `Lightning` aren't normally considered done until two implementations do it, sort of like the way that things do it, call it interop.
```

Sounds like "implementations". WDYT? @0tuedon

> And so I'm working on interop with T-Dash and once we get two Limitations that have it done, then it's officially like, you know, considered launched or whatever.

Reviewer: ```suggestion
And so I'm working on interop with T-Bast and once we get two Limitations that have it done, then it's officially like, you know, considered launched or whatever.
```

In https://btctranscripts.com/stephan-livera-podcast/2021-11-07-t-bast ,
it uses `T-Bast`

Gemini:
<img width="1820" height="1006" alt="Image" src="https://github.com/user-attachments/assets/c2b188ef-08a3-414b-8ed0-acd05259fea7" />

Or, consistent with `Tbast` used in many places of this transcript. @0tuedon WDYT?

> And so I'm working on interop with T-Dash and once we get two Limitations that have it done, then it's officially like, you know, considered launched or whatever.

Reviewer: ```suggestion
And so I'm working on interop with T-Dash and once we get two implementations that have it done, then it's officially like, you know, considered launched or whatever.
```
Sounds like "implementations". WDYT?

You guys got any other core projects you guys are favoring?

**Tony Giorgio:** 01:01:57

Some of the transaction v3 stuff to kind of help against paying tax on `Lightning`.

**Bastien Teinturier:** 01:02:03

In the short term, I would really like to see package relay and ephemeral anchors, because that's something that's only a policy change.
So in theory-

**NVK:** 01:02:11

Package relay shouldn't be hard to change.
It's not consensus.

**Bastien Teinturier:** 01:02:15

Yeah, so it should be easier to train.
It's actually, people are working on that.
It's making progress, but like everything, it takes time because you want to make sure you don't create new DOS issues and you really fix the problems you are trying to fix.
But this would have had a huge impact on `Lightning`, especially nowadays with the high fee situation, because once we have that, we can actually make the commitment transactions pay zero fees and only pay fees by doing CPFP, which means that when you are touching the chain, you see exactly why you have to pay fees and you can choose that fee rate.
But then once you're in `Lightning`, you are completely decorrelated from unchanged fees.

> Whereas nowadays, this is not the case because since the commitment transaction needs to pay a fee that has to be able to at least enter the mempool, whenever you add HCLCs to a channel, you have to take into account the fact that you are the commitment transaction is getting bigger.

Reviewer: ```suggestion
Whereas nowadays, this is not the case because since the commitment transaction needs to pay a fee that has to be able to at least enter the mempool, whenever you add HTLCs to a channel, you have to take into account the fact that you are the commitment transaction is getting bigger.
```

How does it leverage APO?
Like the change you were mentioning?

**Dusty Dettmer:** 01:04:17

Oh, APO is so exciting.
Well, I mean, the big thing about APO is that the `Lightning` database is going to go a smaller, which isn't necessarily an on-chain thing, but like currently, whenever you do a `Lightning` transaction, What we're really doing behind the scenes is we're making a whole bunch of Bitcoin transactions that we sign and just never actually publish.
We end up with this, if you do thousands of payments, each one of those payments is a whole collection of transactions we made to go along with that.
You end up with tens of thousands, hundreds of thousands of Bitcoin, unpublished Bitcoin transactions.
With APO, all of that goes away.
The big thing is because when you chain multiple transactions together, like I published this transaction with spends into this transaction to that one, ABC, all child of each other, they have to be signed referencing who their parent is, right?
And APL allows us to separate that bond, be like this child transaction, we can sign it for any parent at all.

> And that just makes, it makes `Lightning` just so much better in general, but it also makes things like wash showers a lot better.

Reviewer: ```suggestion
And that just makes, it makes `Lightning` just so much better in general, but it also makes things like watchtowers a lot better.
```

https://www.youtube.com/watch?v=SjLPbjs9LkI&t=3914s

It just it just fixes the entire like everything basically.
And to me, I don't think it's I think it's not a controversial change, which is maybe why it's not getting a lot of attention.
It's just there's nothing wrong with it.
It's sort of an obvious like next of improvement for Bitcoin in my eyes, I haven't seen a good critique of it.

**NVK:** 01:05:34

Famous last words.

**Vivek:** 01:05:36

> Yeah, I was just about to say like what Dusty is referring to is, is LN symmetry, previously known as L2, and it has a channel factory benefits and things of that nature.

Reviewer: ```suggestion
Yeah, I was just about to say like what Dusty is referring to is, is LN-Symmetry, previously known as Eltoo, and it has a channel factory benefits and things of that nature.
```

https://bitcoinops.org/en/topics/eltoo/

So I don't know.
We'll see where that ends up, perhaps.

**NVK:** 01:06:16

I don't know.

> Like I'm partial to CTV and OpVault sort of path there because I really want to see that stuff happen.

Reviewer: ```suggestion
Like I'm partial to CTV and OP_VAULT sort of path there because I really want to see that stuff happen.
```

I think you can do some of that stuff with CTV instead of APO.

**Vivek:** 01:06:27

> You cannot, you need another soft fork for a check

Reviewer: ```suggestion
You cannot, you need another soft fork for CHECKSIGFROMSTACK
```

Speaker 2: 01:06:30

to

**Vivek:** 01:06:30

sig from stack.

> 

Reviewer: Delete. These are part of L1273 `CHECKSIGFROMSTACK`

**NVK:** 01:06:31

That's right.

**Vivek:** 01:06:32

So it gets very hairy about like, who wants which soft fork for which purpose?
Cause the covenants, yeah.

**NVK:** 01:06:37

Oh, it's going to be like, first it's going to be very hard to get any soft fork done in the next few years.
Like realistically speaking, we can't even agree on how to activate it, which is even more contentious than the actual content.

**Jeff Czyz:** 01:06:49

Yeah.

> And when we build on those, L2 is the name I was familiar with.

Reviewer: ```suggestion
And when we build on those, Eltoo is the name I was familiar with.
```

> Is there a new name for L2 right now?

Reviewer: ```suggestion
Is there a new name for Eltoo right now?
```

**Vivek:** 01:06:56

> Yeah, I believe InstaGibs has taken the mantle and He's calling it LN Symmetry now.

Reviewer: ```suggestion
Yeah, I believe Instagibbs has taken the mantle and He's calling it LN-Symmetry now.
```

And this is one of the biggest things that we're interested in the most when we actually get splicing and we can use it, is the concept of being able to do coin joins from splicing rounds.
I mean, not just for privacy standpoint, but also a scalability standpoint, too.
I mean, I think some of the free rate stuff kind of shows that, you know, we sort of need collaborative transactions to be able to help scale Bitcoin on chain usage even more.
And part of that is even channel open, channel opens.
So being able to do coin join in channel opens, dealing with, you know, the toxic change, not having that privacy leak, that's some of the things that we're kind of wanting to build into LN4TX, which is something Ben Carman started on specifically to do CoinJoins on `Lightning`.
I think splicing really changes the game from what we can do, because also it's like zero downtime, too.
So you can say if you can add CoinJoin liquidity onto the `Lightning` network, you're still able to route and collect yield off of your routing fees.
You now just basically created a way to coin join and make fees at the same time.
That's huge.
Let's go.

**NVK:** 01:19:38

So like here's the thing, right.

> So, for example, joint market, unfortunately, like kind of like it has volume that kind of comes and goes.

Reviewer: ```suggestion
So, for example, JoinMarket, unfortunately, like kind of like it has volume that kind of comes and goes.
```

How do you guys feel about it?
What's going on with the heuristics?

**Dusty Dettmer:** 01:26:20

Oh man, splicing is more important than ever with all the spam going on.
Right.
Like I think I think

**NVK:** 01:26:25

they're valid.
Shitcoin transactions.
There is no such thing as spam in Bitcoin.
I will repeat this 50 times.
If it's a valid transaction is a valid transaction.

**Dusty Dettmer:** 01:26:38

I think it's a weird it's a weird match for for Spicing where it's like I think people are paying more attention to Spicing now because of all the legitimate on-chain activity.

**NVK:** 01:26:49

> DGN complete retard activity.

Reviewer: ```suggestion
Degen complete retard activity.
```

But

**NVK:** 01:26:53

it's still valid transactions.

**Dusty Dettmer:** 01:26:57

Oh man.

> But it's kind of funny that the bedfellow of getting Spice and getting more attention is these DGNs. Doing whatever they're doing on-chain, which the more I look into it, the more I'm like, what am I looking at?

Reviewer: ```suggestion
But it's kind of funny that the bedfellow of getting Spice and getting more attention is these degens. Doing whatever they're doing on-chain, which the more I look into it, the more I'm like, what am I looking at?
```

So can you just imagine like, it's a great dry run in my opinion, for us to sort of like get a little taste of what it's like when there is not just price, Because when the price goes up, you know, all the retail comes in, they all want to buy their, you know, their 615 Bitcoin.
But like now we have other people coming in with all these other users that, you know, are completely yotic.
But they do set a floor, right, for transactions.
And, you know, `Lightning` has to get its shit together.
Right.
Because `Lightning` was never in scale around a high fee market.

**Dusty Dettmer:** 01:28:54

> I think this brings up a great point of if you look at there's some apps like the Moon Wallet, maybe people will try it.

Reviewer: ```suggestion
I think this brings up a great point of if you look at there's some apps like the Muun Wallet, maybe people will try it.
```

> M U U and Moon Wallet, where they're they're trying to be this wallet where you can pay to an on-chain destination and also to a `Lightning` destination, right?

Reviewer: ```suggestion
M U U and Muun Wallet, where they're they're trying to be this wallet where you can pay to an on-chain destination and also to a `Lightning` destination, right?
```

But their concept is storing all the Bitcoin on-chain and moving it to `Lightning` on demand.
That works fine when on-chain fees are like 10 cents, but now that they're like $10, suddenly every `Lightning` payment, which should be basically free, costs $10 on Moon.

> So that kind of app isn't sustainable, but splicing enables you to make a moon wallet in the reverse.

Reviewer: ```suggestion
So that kind of app isn't sustainable, but splicing enables you to make a Muun wallet in the reverse.
```

But eventually, I think all of them are gonna have to switch over to it.
And this is gonna be a big UX improvement for new users.

**NVK:** 01:30:10

> I still remember on Breeze where I didn't read the fine print, I never read the fine print.

Reviewer: ```suggestion
I still remember on Breez where I didn't read the fine print, I never read the fine print.
```

Reviewer: There are other ~~Breeze~~ typos

And it's okay if we have to change it entirely afterwards because it's broken.
So it's a good ground for experimentation, but we don't want to waste too much time either on experimenting with too many things that we know won't work in the long run.

**NVK:** 01:40:16

There you go.
So see if you are a researcher who can make a production grade cryptographic library apply for a grant, you know, that's what OpenSats is for.

**Vivek:** 01:40:27

> Just please focus on MuSig too.

Reviewer: ```suggestion
Just please focus on MuSig2.
```

Forget about all this ZKP stuff.

**NVK:** 01:40:31

I am done in MuSig too.

> I am a Frost evangelist now.

Reviewer: ```suggestion
I am a FROST evangelist now.
```

Reviewer: https://glossary.blockstream.com/frost/

`Frost` should be replaced with `FROST` in many other places

A fully Frost sold Frost maximalist.

**Dusty Dettmer:** 01:40:38

> Are you Roast sold though?

Reviewer: ROAST: Robust Asynchronous Schnorr Threshold Signatures
https://blog.blockstream.com/roast-robust-asynchronous-schnorr-threshold-signatures/

> Are you Roast sold though?

Reviewer: ```suggestion
Are you ROAST sold though?
```

**NVK:** 01:40:39

> I mean, the Roast can get interested.

Reviewer: ```suggestion
I mean, the ROAST can get interested.
```

That's right.

Speaker 2: 01:47:03

So

**Tony Giorgio:** 01:47:04

it's like you have to do all the policy stuff.
And maybe you can still do that.
You know, you can have, I don't know, some secure channel that that it talks to with with your actual running node.

## FROST

**NVK:** 01:47:15

I mean,

**Tony Giorgio:** 01:47:15

it's hard

**NVK:** 01:47:16

once we have frost that it changes a bit.
Right.
Because now you can have some better thresholds distribution, right?
Of how you move funds, the authorization of the funds.
And those signatures can actually move probably through the `Lightning` Message Network as well.
I don't know.
It's like this is all like sort of like moon brain for later.

**Jeff Czyz:** 01:47:38

Yes.
Let's

**NVK:** 01:47:41

add Frost to splicing.
Let's go.
That's right.

**Tony Giorgio:** 01:47:45

Let's go.
Let's do `Lightning` communication.
Let's do it on Nostr.

Speaker 2: 01:47:48

Let's switch.

**Vivek:** 01:47:48

Frost splices on Nostr in two weeks.
You heard it here first.
That's right.

**NVK:** 01:47:52

That's right.
Okay, guys, any other things that should be part of this conversation that we missed?

**Jeff Czyz:** 01:47:59

> Well, I don't know if we touched on this, but there is that swap in potentium proposal.

Reviewer: ```suggestion
Well, I don't know if we touched on this, but there is that Swap-in-Potentiam proposal, the ZmnSCPxj, which I believe is splicing. I'm only vaguely familiar with it. I've read it a little while back.
```

Reviewer: https://www.youtube.com/watch?v=SjLPbjs9LkI&t=6483s

## ecash

**NVK:** 01:48:41

> You know, like when the Galaxy brain gets a little too far, I go, I go, I slide into Rindell's DMs. So, okay, so I think there's one more topic that I'd love to address is the e-cash stuff on top of `Lightning` that like, to me, it fixes pretty much most of the scalability problems and it makes the custodials to be 10 times better, even though there's two custodials.

Reviewer: ```suggestion
You know, like when the Galaxy brain gets a little too far, I go, I go, I slide into Rijndael's DMs. So, okay, so I think there's one more topic that I'd love to address is the e-cash stuff on top of `Lightning` that like, to me, it fixes pretty much most of the scalability problems and it makes the custodials to be 10 times better, even though there's two custodials.
```

But if you do it correctly, and you're willing to give up some risk, lots of funds, it's a huge improvement from a privacy perspective.
Now being able to receive into a mint, now you've just effectively got receiver privacy.
And then being able to send that out at any time, at least waiting like a day or so, you send that out and the custodian doesn't know who the original person was.
They don't even know where the source of funds came from, first of all, but then they can't correlate this incoming payment is now correlated to this outgoing payments on `Lightning`.
It's just a huge step up in privacy.

**NVK:** 01:50:41

You know, one of the reasons why I bring this up is not because it's just the next new shiny thing.

> It's, It's because like, at least to me, like, I really believe that between Nostra and eCash is like where most of the `Lightning` demand is going to come in the next couple of years.

Reviewer: ```suggestion
It's, It's because like, at least to me, like, I really believe that between Nostr and eCash is like where most of the `Lightning` demand is going to come in the next couple of years.
```

Reviewer: Probably more `Nostr` typos in five other places.

I don't think it's going to be from people, normies trying to buy coffee.
I think it's going to be from this sort of like side technologists trying to address some problems with like very specific solutions.

> So like, it's going to be like, you know, people trying to do mining Fetty pools, right?

Reviewer: ```suggestion
So like, it's going to be like, you know, people trying to do mining Fedi pools, right?
```

> So like they're Fetty Mint pools for mining.

Reviewer: ```suggestion
So like they're Fedimint pools for mining.
```

It's going to be, you know, this kind of like very specific thing that requires the e-cash solution in a way that they're developing.
And it's gonna bring like huge demand for `Lightning`.

**Jeff Czyz:** 01:51:31

It also really touches on the scalability aspect of Bitcoin.
And like you mentioned earlier, not everyone will have a UTXO.

> And the Feddy Mintz model where there's sort of these community banks, if you will, and interoperability through `Lightning` helps solve that.

Reviewer: ```suggestion
And the Fedimint model where there's sort of these community banks, if you will, and interoperability through `Lightning` helps solve that.
```

And like, you know, and this sort of community aspect is also about, you know, how would you onboard people?
Like if everyone's not going to have a UTXO, well, they're going to have to earn Bitcoin, right?
And I really think that's a pretty elegant solution.
Hopefully you see this.

**NVK:** 01:52:04

I mean, like I was, you know, I always found eCash to be sort of like, kind of like idiotic at like how it was sort of like, like presented and available back in the day, because you never had like an anchor of truth in it.
So, you know, it was kind of like, you know, might as well just use like Fiat.
You know, now that we have like `Lightning` sort of being the IO for eCash and that be the source of truth for that sort of mint, like it changes everything, right?
Like it makes the thing like fully usable.

> And it does, again, it breaks down those ETXOs into smaller units and do not have a footprint on chain until you try to get out of that mint.

Reviewer: ```suggestion
And it does, again, it breaks down those UTXOs into smaller units and do not have a footprint on chain until you try to get out of that mint.
```

I got `Lightning`splice.com and the goal is to put all the resources that help try to explain things to people there in one spot.
So check that out `Lightning`splice.com.

**NVK:** 01:55:17

Very cool.
By the way, guys, if anybody has any resources, links or whatever, feel free to just put on the on the chat and we're going to add to the show notes.
Vivek, any final thoughts?

**Vivek:** 01:55:29

Yeah, I just want to say thank you to all of our guests and everyone that's implementing splices.
I think full credit to Dusty on the CLN side.
I think Lisa as well on the `LDK` side, I believe that's Jeff and maybe someone else.
Jeff, you can chime in on who.

**Jeff Czyz:** 01:55:46

Yeah.
So Dodkin's doing some of the permitting work, also fuel funding.
And then Jervis and myself will be our sort of working on splicing, but early stages.

**Vivek:** 01:55:56

Awesome.

**Dusty Dettmer:** 01:55:56

So you're the splicing guy over there.
Awesome.
Yeah.

**Jeff Czyz:** 01:55:59

Yeah.
So next next quarter, Q3 is the goal for sometime Q3 to have it done.
Let's go.

**NVK:** 01:56:04

Look at that.

**Vivek:** 01:56:05

> And then thank you to T-Best as well, who's tested the early interop for these sort of things with like V2 opens and whatnot.

Reviewer 2: ```suggestion
And then thank you to ⁠Tbast as well, who's tested the early interop for these sort of things with like V2 opens and whatnot.
```

Really grateful for that.

> And I believe on the L&D side that someone named Eugene correct me if I'm mistaken but yeah I want to say thank you because there's just too much crap now like we talked about APO, LN symmetry with splices and with music too and then incorporating that into payjoins and then how does that tie in with like other routing pathfinding stuff for mobile with async payments and trampoline, you know?

Reviewer: ```suggestion
And I believe on the L&D side that someone named Eugene correct me if I'm mistaken but yeah I want to say thank you because there's just too much crap now like we talked about APO, LN-Symmetry with splices and with MuSig2 and then incorporating that into payjoins and then how does that tie in with like other routing pathfinding stuff for mobile with async payments and trampoline, you know?
```
